server/controllers/auth/login.js

```diff
@@ -1,3 +1,31 @@
-exports.login = (req, res) => {
-  res.send(' Hello from login file');
+const { compare } = require('bcrypt');
+const { jwtSign } = require('../../helpers');
+const { getUserByUsername } = require('../../database/queries/users');
+
+exports.login = (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) next({ code: 400, msg: 'bad request !!!!' });
+
+  const key = process.env.KEY;
+  let id;
+  getUserByUsername(username)
+    .then(({ rows }) => {
+      id = rows[0].id;
+      return compare(password, rows[0].password);
```

Comment:
I think you need to hash the password first

Reply:
Done

Reply:
the password in the database is stored hashed. Thus to compare them you must hash the password from the request first